cosmwasm/enclaves/shared/contract-engine/src/contract_validation.rs

```diff
@@ -351,100 +357,123 @@ pub fn validate_basic_msg(
 }
 
 /// Verify all the parameters sent to the enclave match up, and were signed by the right account.
-#[allow(unused_variables)]
+#[allow(clippy::too_many_arguments)]
 pub fn verify_params(
     sig_info: &SigInfo,
     sent_funds: &[Coin],
     sender: &CanonicalAddr,
     contract_address: &HumanAddr,
     msg: &SecretMessage,
-    og_msg: &[u8],
+    #[cfg(feature = "light-client-validation")] og_msg: &[u8],
+    should_validate_sig_info: bool,
+    should_validate_input: bool,
+    handle_type: HandleType,
 ) -> Result<(), EnclaveError> {
-    debug!("Verifying message signatures for: {:?}", sig_info);
-
-    //let start = Instant::now();
-    // If there's no callback signature - it's not a callback and there has to be a tx signer + signature
-    if let Some(callback_sig) = &sig_info.callback_sig {
-        return verify_callback_sig(callback_sig.as_slice(), sender, msg, sent_funds);
-    }
-    // let duration = start.elapsed();
-    // trace!(
-    //     "verify_params: Time elapsed in verify_callback_sig: {:?}",
-    //     duration
-    // );
-    #[cfg(feature = "light-client-validation")]
-    if !check_msg_matches_state(og_msg) {
-        return Err(EnclaveError::ValidationFailure);
+    if should_validate_sig_info {
```

Comment:
If you're adding an if to 80% of a function's logic, it's a sign that the code should be arranged differently

Reply:
https://github.com/scrtlabs/SecretNetwork/commit/d91cb2179